lib/remote_system.rb

```diff
@@ -117,7 +127,7 @@ def retrieve_files(filelist, destination)
 
     cmd = [
       "rsync",
-      "-e", "ssh",
+      "-e", "ssh -p #{@ssh_port}#{' '+@ssh_key_options.join(' ') if @ssh_key_options.length > 0}",
```

Comment:
Surrounding space missing for operator `+`.<br>Prefer double-quoted strings inside interpolations.